### PR TITLE
minor: small cleanup

### PR DESCRIPTION
**Pull request by kkrolczyk**
cleanup of translator.py and project.py
no functionality has been changed, just removed some duplication ect

Signed-off-by: Krzysztof Królczyk <Krzysztof.Krolczyk@o2.pl>

### CODE REVIEW

File: flict/flictlib/translator.py [hunk 1]
```diff
@@ -12,12 +12,20 @@
 
 import argparse
 import json
-from flict.flictlib import logger
 from argparse import RawTextHelpFormatter
+try:
+    from flict.flictlib import logger as logger
+except ImportError: # allows usage also as a standalone tool
+    class MonkeyPatchLogger:
```
Comment (priv-kweihmann) [inline flict/flictlib/translator.py]: That deserve some explanation, as I don't quite follow the given info

File: flict/flictlib/translator.py [hunk 2]
```diff
@@ -61,23 +69,19 @@ def parse():
                         dest='translation_file',
                         help='files with translations, default is ' + DEFAULT_TRANSLATIONS_FILE,
                         default=DEFAULT_TRANSLATIONS_FILE)
-    parser.add_argument('-p', '--package-file',
-                        type=str,
-                        dest='package_file',
-                        help='file with package definition')
+#    parser.add_argument('-p', '--package-file',
```
Comment (priv-kweihmann) [inline flict/flictlib/translator.py]: Why disable them? Could you kindly provide some insights. Even if they're used in the current setup it's worth making the removal a separate commit

File: flict/flictlib/translator.py [hunk 3]
```diff
@@ -94,61 +98,68 @@ def update_license(translations, license_expr):
...
-                    transl = item['spdx_id']
-                    key = item['license_expression']
-                    if transl not in symbols:
-                        symbols[transl] = []
-                    symbols[transl].append(key)
+    try:
+        with open(translations_file) as file_:
+            translations_object = json.load(file_)
+    except (FileNotFoundError, json.JSONDecodeError):
+        raise RuntimeError("reading translations file failed")
```
Comment (priv-kweihmann) [inline flict/flictlib/translator.py]: We are using our own Exception set for all kinds of exception to encountered - please have a look at https://github.com/vinland-technology/flict/blob/e2a15bdcc59d3de72558d817a26e6f4ef5344a94/flict/flictlib/return_codes.py#L50 and turn it into a FlictError with appropriate error code

File: flict/flictlib/translator.py [hunk 4]
```diff
@@ -94,61 +98,68 @@ def update_license(translations, license_expr):
...
+                logger.main_logger.debug(" IGNORING since scancode")
             else:
-                raise Exception("Failed parsing item:  " + str(item))
+                transl = item['spdx_id']
+                key = item['license_expression']
+                if transl not in symbols:
+                    symbols[transl] = []
+                symbols[transl].append(key)
+        else:
+            raise RuntimeError(f"Failed parsing item: {item}")
```
Comment (priv-kweihmann) [inline flict/flictlib/translator.py]: Same here

File: flict/flictlib/translator.py [hunk 5]
```diff
@@ -94,61 +98,68 @@ def update_license(translations, license_expr):
...
-        deps = package["dependencies"]
-        license = package["license"].strip()
-        package["license"] = update_license(translations, license)
-        update_packages(translations, deps)
-        return packages
+    try:
+        with open(jsonfile) as file_:
+            packages = json.load(file_)
+    except (FileNotFoundError, json.JSONDecodeError):
+        raise RuntimeError("reading packages file failed")
```
Comment (priv-kweihmann) [inline flict/flictlib/translator.py]: and here once more

File: flict/flictlib/translator.py [hunk 6]
```diff
@@ -94,61 +98,68 @@ def update_license(translations, license_expr):
...
-        package = packages["component"]
-        deps = package["dependencies"]
-        license = package["license"].strip()
-        package["license"] = update_license(translations, license)
-        update_packages(translations, deps)
-        return packages
+    try:
+        with open(jsonfile) as file_:
+            packages = json.load(file_)
+    except (FileNotFoundError, json.JSONDecodeError):
```
Comment (priv-kweihmann) [inline flict/flictlib/translator.py]: I personally would also cover PermissionError here in case the user doesn't have read access to the file

File: flict/flictlib/translator.py [hunk 7]
```diff
@@ -94,61 +98,68 @@ def update_license(translations, license_expr):
...
+    # TODO: sync with flict (should be "package")
+    package = packages["component"]
+    deps = package["dependencies"]
+    license = package["license"].strip()
+    package["license"] = update_license(translations, license)
+    update_packages(translations, deps)
+    return packages
+
+
+if __name__ == "__main__":
```
Comment (priv-kweihmann) [inline flict/flictlib/translator.py]: I see that as a change of functionality - so it should come as a separate commit with some justification

**Review by priv-kweihmann — CHANGES_REQUESTED** [review]
See comments

File: flict/flictlib/translator.py (hunk 1 above)
Comment (kkrolczyk) [reply flict/flictlib/translator.py]: When running the file stand-alone, it would try to import logger and fail, provided the script could be executed from the same directory where it is located. There aren't many dependencies, actually none, except for the logger. By creating mocked logger class, the error would be circumvented, while preserving original functionality (code is imported in flict/flictlib/license.py, that seems to be the only usage of the code from that file)

File: flict/flictlib/translator.py (hunk 2 above)
Comment (kkrolczyk) [reply flict/flictlib/translator.py]: I actually got an opposite impression. Please look at the comment above. Actual usage is currently limited to importing single function from that file, however there was also argparse, as it would have been intended to be used separately, from command line. However, `parse()` was never called, in case the file was executed from command line. As i didn't know the history, and more importantly the reason or idea behind that - instead of removing them i thought it might be more beneficial to keep them commented out - after enabling proper parsing from command line, an user could be surprised reading help, that graph or packaging file could be provided to the script, while in fact they weren't used, nor the code for graph creation was there. So, long story short - i thought this would prevent confusing user, prevent misleading.

File: flict/flictlib/translator.py (hunk 3 above)
Comment (kkrolczyk) [reply flict/flictlib/translator.py]: indeed, i have seen this in other files. However, this would add another dependency to be resolved during importing symbol. Perhaps this file was never meant to be a stand-alone script, which could be used separately?
I could use RET_FILE_NOT_FOUND and RET_INVALID_EXPRESSSION - but i think it was purposely just for invalid license expression. Should one create a new return code, for general errors in input JSON file?

File: flict/flictlib/translator.py (hunk 7 above)
Comment (kkrolczyk) [reply flict/flictlib/translator.py]: I'd say this is a bit of a gray area. As explained in the first comment, the file would would just add a different way to work with the file, without modifying how it was used from within flict. However, probably it might be more important to understand if this was ever the case, or could it be useful to work with it bypasing flict? Nevertheless - it could be of course reworked to be a separate commit, if it was actually useful to to have.

Comment (priv-kweihmann) [reply flict/flictlib/translator.py]: @hesa could you provide some insights if this was ever meant to run standalone?

File: flict/flictlib/translator.py (hunk 1 above)
Comment (hesa) [reply flict/flictlib/translator.py]: Is there a use case for running this file stand alone?

Comment (kkrolczyk) [reply flict/flictlib/translator.py]: That's a great question. I don't think so - but perhaps there is a different work-flow where someone would benefit from it? If not - perhaps this, as well as `argparse` block could be removed instead.

**Comment by hesa** [comment]
I would rather we remove the stand alone functionality from all files (apart from the one "real" main file).

**Comment by kkrolczyk** [comment]
> I would rather we remove the stand alone functionality from all files (apart from the one "real" main file).

Sure, done :)

File: flict/flictlib/translator.py [hunk 8]
```diff
@@ -94,61 +28,62 @@ def update_license(translations, license_expr):
...
+                if transl not in symbols:
+                    symbols[transl] = []
+                symbols[transl].append(key)
+        else:
+            raise FlictError(ReturnCodes.RET_INVALID_PROJECT,
+                             f"Failed parsing item: {item} in translations.")
+    results = {'original': translations_object,
+               'translations_map': translations_map,
+               'symbols': symbols}
+    # enabling this crashes lots of tests :)
```
Comment (priv-kweihmann) [inline flict/flictlib/translator.py:69]: What tests are we talking about?

**Comment by jens-erdmann** [comment]
This is no longer a small cleanup. The commit message doesn't reflect on the actual changes anymore. Cramming all these changes into one commit also doesn't sit well with me. Can that be split up a bit?

**Comment by kkrolczyk** [comment]
@jens-erdmann  sure, so far i see 2 major things:
- cosmetics in project.py and relicense.py
- removal of unused parse() and duplicated code inside translator.py - probably this would be the split point


@priv-kweihmann  failing tests

```diff --git a/flict/flictlib/translator.py b/flict/flictlib/translator.py
index 99fcbc2..8d370fe 100755
--- a/flict/flictlib/translator.py
+++ b/flict/flictlib/translator.py
@@ -67,9 +67,9 @@ def read_translations(translations_file):
                'translations_map': translations_map,
                'symbols': symbols}
     # enabling this crashes lots of tests :)
-    # return results
-    read_translations.results = results
-    return symbols
+    return results
+    # read_translations.results = results
+    # return symbols
```

ERROR: tests/test_outbounds.py::test_outbound
ERROR: tests/test_verify.py::test_verify_file
ERROR: tests/test_outbounds.py::test_no_relicense
ERROR: tests/test_verify.py::test_verify_expression
ERROR: tests/test_report.py::ReportTest::test_outbound
FAIL: tests/test_project.py::Basic::test_small
FAIL: tests/test_compat.py::test_compat
FAIL: tests/test_project.py::Basic::test_big
FAIL: tests/test_simplified.py::test_simplify
FAILED (errors=6, failures=4)

tracedumps are quite big, so i'll skip posting them here. Given the current tests  

some are quite confusing:
```
Exception: BSD is not a supported license.
During handling of the above exception, another exception occurred:
...
flict.flictlib.return_codes.FlictError: (<ReturnCodes.RET_INVALID_EXPRESSSION: (11, 'Invalid expression')>, "Could not parse license expression: ['MIT BSD GPL-2.0-only WITH Classpath-exception-2.0']")
```

so basically, i think some tests assume that single "symbols" are returned from the read_translations(translations_file) function.

I've done a quick search for usage of functions from `translator.py` file:
- read_packages_file() is unused
- read_translations() is used by `license.py`
- update_{packages,license} are used only within `translator.py`
- this includes searching in tests directory

**Comment by priv-kweihmann** [comment]
> @jens-erdmann sure, so far i see 2 major things:
> 
> * cosmetics in project.py and relicense.py
> * removal of unused parse() and duplicated code inside translator.py - probably this would be the split point
> 
> @priv-kweihmann failing tests
> 
> ```diff
> index 99fcbc2..8d370fe 100755
> --- a/flict/flictlib/translator.py
> +++ b/flict/flictlib/translator.py
> @@ -67,9 +67,9 @@ def read_translations(translations_file):
>                 'translations_map': translations_map,
>                 'symbols': symbols}
>      # enabling this crashes lots of tests :)
> -    # return results
> -    read_translations.results = results
> -    return symbols
> +    return results
> +    # read_translations.results = results
> +    # return symbols
> ```
> 
> ERROR: tests/test_outbounds.py::test_outbound ERROR: tests/test_verify.py::test_verify_file ERROR: tests/test_outbounds.py::test_no_relicense ERROR: tests/test_verify.py::test_verify_expression ERROR: tests/test_report.py::ReportTest::test_outbound FAIL: tests/test_project.py::Basic::test_small FAIL: tests/test_compat.py::test_compat FAIL: tests/test_project.py::Basic::test_big FAIL: tests/test_simplified.py::test_simplify FAILED (errors=6, failures=4)
> 
> tracedumps are quite big, so i'll skip posting them here. Given the current tests
> 
> some are quite confusing:
> 
> ```
> Exception: BSD is not a supported license.
> During handling of the above exception, another exception occurred:
> ...
> flict.flictlib.return_codes.FlictError: (<ReturnCodes.RET_INVALID_EXPRESSSION: (11, 'Invalid expression')>, "Could not parse license expression: ['MIT BSD GPL-2.0-only WITH Classpath-exception-2.0']")
> ```
> 
> so basically, i think some tests assume that single "symbols" are returned from the read_translations(translations_file) function.
> 
> I've done a quick search for usage of functions from `translator.py` file:
> 
> * read_packages_file() is unused
> * read_translations() is used by `license.py`
> * update_{packages,license} are used only within `translator.py`
> * this includes searching in tests directory

Ok - guess that looks like another refactoring incoming :disappointed: -- @hesa FYI (as I guess that one is one you)
@kkrolczyk I'm okay with solving it like it is right now (but as Jens already suggested it should come as a separated commit)

File: flict/flictlib/translator.py (hunk 8 above)
Comment (kkrolczyk) [reply flict/flictlib/translator.py:69]: replied below
